azure/types.go

```diff
@@ -407,6 +410,18 @@ type LoadBalancerProfile struct {
 	IdleTimeoutInMinutes *int32
 }
 
+// APIServerAccessProfile is the access profile for AKS API server.
+type APIServerAccessProfile struct {
+	// AuthorizedIPRanges - Authorized IP Ranges to kubernetes API server.
+	AuthorizedIPRanges []string
+	// EnablePrivateCluster - Whether to create the cluster as a private cluster or not.
+	EnablePrivateCluster *bool
+	// PrivateDNSZone - Private dns zone mode for private cluster.
+	PrivateDNSZone *string
+	// EnablePrivateClusterPublicFQDN - Whether to create additional public FQDN for private cluster or not.
+	EnablePrivateClusterPublicFQDN *bool
```

Comment:
I believe this will default to true on AKS side if private cluster is enabled. should we let that happen, or default it inside CAPZ?

it would only mean that a user who explicitly sets it would trigger a no-op reconcile, and a user who wants to change it will be able to.

Reply:
Thanks for the feedback! I would slightly prefer letting AKS to handle the default. My rationales are:
1. This is true for other fields as well (e.g. [managedOutboundIPs](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/exp/api/v1alpha4/azuremanagedcontrolplane_types.go#L127), [sku tier](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/exp/api/v1alpha4/azuremanagedcontrolplane_types.go#L115))
2. We need to add some logic when choosing the default values (e.g. set `EnablePrivateClusterPublicFQDN` to true when `EnablePrivateCluster` is true)
3. Given 1,2, this will add some extra complexity to the CAPZ code
The downside is that there will be some unnecessary no-op reconciles, but i feel that's probably fine?

I'm open to change it if you feel strongly about it :)

Reply:
Fine w me, I asked because I know this defaulting too true has surprised some folks in the past

Reply:
But that’s really an aks issue